### PR TITLE
Target 0.44.0: Fix osx wheel build

### PR DESCRIPTION
**Pull request by sklam**
Target branch is `release0.44`.
Needs to be cherrypicked later to main

Summary:
- For OSX wheel, the llvmdev manylinux recipe is used to remove `zstd` dependency.
- ~Additionally, building of compiler-rt is turned off. The build machines do not have iOS SDK, the missing SDK is causing errors. The fix to not error-out due to iOS SDK is introduced in llvm 17 (https://github.com/llvm/llvm-project/commit/78fb02107ae5819501af2600b0bebd48c1900a9b)~ turned off iOS support and renabled compiler-rt in https://github.com/numba/llvmlite/pull/1104/commits/135be02cd38b0485dfb2ee55093a54aae803ef85
- Other changes is to un-comment-out the OSX specifics in meta.yaml

### CODE REVIEW

File: conda-recipes/llvmdev_manylinux/build.sh [hunk 1]
```diff
@@ -62,7 +62,7 @@ cmake -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
       -DLLVM_ENABLE_Z3_SOLVER=OFF \
       -DLLVM_OPTIMIZED_TABLEGEN=ON \
       -DCMAKE_POLICY_DEFAULT_CMP0111=NEW \
-      -DCOMPILER_RT_BUILD_BUILTINS=ON \
+      -DCOMPILER_RT_BUILD_BUILTINS=OFF \
```
Comment (gmarkall) [inline conda-recipes/llvmdev_manylinux/build.sh:65]: Would 

```suggestion
      -DCOMPILER_RT_BUILD_BUILTINS=ON \
      -DCOMPILER_RT_ENABLE_IOS=OFF \
```

work around the issue and enable compiler-rt to be built?

Comment (sklam) [reply conda-recipes/llvmdev_manylinux/build.sh]: The suggestion worked. Compiler-rt is reenabled in https://github.com/numba/llvmlite/pull/1104/commits/135be02cd38b0485dfb2ee55093a54aae803ef85

**Comment by gmarkall** [comment]
To clarify my understanding - is the llvmdev conda package used on Windows and Linux, and only the manylinux one recipe used on macOS?

**Review by gmarkall — APPROVED** [review]
Build / recipe changes look fine to me (caveat being my limited knowledge in this area!)

**Comment by sklam** [comment]
RE: https://github.com/numba/llvmlite/pull/1104#issuecomment-2504784045

`llvmdev_manylinux` recipe is used by macOS and Linux wheels
`llvmdev` recipe is used by Windows.

**Comment by gmarkall** [comment]
Noticed this wasn't in the 0.44 milestone so I've just popped it in there to make sure it doesn't get missed.

**Comment by sklam** [comment]
Turns out we need a zlib fix for windows. I'll open a follow up PR base on this one.

**Comment by esc** [comment]
Note: This will need a backport to the `release0.44` branch once merged.

**Comment by esc** [comment]
> Note: This will need a backport to the `release0.44` branch once merged.

Correction, this will need a backport to `main`.

**Comment by esc** [comment]
Backport (or maybe forward port) is here: https://github.com/numba/llvmlite/pull/1113

**Pull request merged**